bookkeeper-dist/all/build.gradle

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+plugins {
+    id 'java'
+    id 'distribution'
+}
+
+description = 'Apache BookKeeper :: Dist (All)'
+
+dependencies {
+
+    implementation project(':bookkeeper-benchmark')
+    implementation project(':bookkeeper-http:http-server')
+    implementation project(':bookkeeper-http:vertx-http-server')
+    implementation project(':bookkeeper-server')
+    implementation project(':bookkeeper-stats-providers:codahale-metrics-provider')
+    implementation project(':bookkeeper-stats-providers:prometheus-metrics-provider')
+    implementation project(':bookkeeper-stats')
+    implementation project(':stream:distributedlog:core')
+    implementation project(':stream:server')
+    implementation project(':bookkeeper-tools')
+
+    compileOnly depLibs.lombok
+    compileOnly depLibs.spotbugsAnnotations
+
+    implementation depLibs.slf4jLog4j
+
+
+    testCompileOnly depLibs.lombok
+
+
+    annotationProcessor depLibs.lombok
+    testAnnotationProcessor depLibs.lombok
+}
+
+jar {
+    archiveBaseName = 'bookkeeper-dist-all'
+}
+
+def depLicences = [
+        "checker-qual-3.5.0/LICENSE",
+        "google-auth-library-credentials-0.20.0/LICENSE",
+        "javax.servlet-api-4.0.0/CDDL+GPL-1.1",
+        "bouncycastle-1.0.2/LICENSE.html",
+        "jsr-305/LICENSE",
+        "netty-4.1.50.Final/*",
+        "paranamer-2.8/LICENSE.txt",
+        "protobuf-3.14.0/LICENSE",
+        "protobuf-3.12.0/LICENSE",
+        "scala-library-2.11.7/LICENSE.md",
+        "scala-parser-combinators_2.11-1.0.4/LICENSE.md",
+        "scala-reflect-2.11.8/LICENSE.md",
+        "slf4j-1.7.25/LICENSE.txt",
```

Comment:
Is there any other way to avoid specifying the exact version here, so that we only need to update the version in 1 place?

Reply:
We can add a wildcard for these. But do we want to tie the version of the packaged dependency? That will be a little more involved.

Reply:
Also couldn't see where we are using scala library. So I don't think we should be packaging these licenses. But we should take this up after we move completely to gradle.

Reply:
Another option would be use a https://github.com/jk1/Gradle-License-Report plugin. With this we can automate the license report and also avoid the need to maintain license files.